tools/cubed_sphere_inc_mod.F90

```diff
@@ -0,0 +1,69 @@
+module cubed_sphere_inc_mod
+
+  use tracer_manager_mod, only: get_tracer_index, get_number_tracers, get_tracer_names
+  use field_manager_mod,  only: MODEL_ATMOS
+  use fv_arrays_mod,      only: fv_atmos_type
+  use sim_nc_mod,         only: open_ncfile, get_var5_r4, check_var_exists, close_ncfile
+
+  implicit none
+  type increment_data_type
+    real, allocatable :: ua_inc(:,:,:)
+    real, allocatable :: va_inc(:,:,:)
+    real, allocatable :: temp_inc(:,:,:)
+    real, allocatable :: delp_inc(:,:,:)
+    real, allocatable :: delz_inc(:,:,:)
+    real, allocatable :: tracer_inc(:,:,:,:)
+  end type increment_data_type
+
+  public read_cubed_sphere_inc, increment_data_type
+
+  contains
+
+!----------------------------------------------------------------------------------------
+
+    subroutine read_cubed_sphere_inc(filename, increment_data, Atm)
+      character(*), intent(in)                 :: filename
+      type(increment_data_type), intent(inout) :: increment_data
+      type(fv_atmos_type), intent(in)          :: Atm
+
+      integer           :: isc, iec, jsc, jec, npz, itracer, ntracers, itile
+      character(len=64) :: tracer_name
+      integer           :: ncid, ncerr
+      
+      ! Get various dimensions
+      call get_number_tracers(MODEL_ATMOS, num_tracers=ntracers)
+      isc = Atm%bd%isc
+      iec = Atm%bd%iec
+      jsc = Atm%bd%jsc
+      jec = Atm%bd%jec
+      npz = Atm%npz
+      itile = Atm%tile_of_mosaic
+
+      ! Open increment file
+      call open_ncfile( trim(filename), ncid )
+
+      ! Read increments
+      call get_var5_r4( ncid, 'u_inc', isc,iec, jsc,jec, 1,npz, itile, 1, increment_data%ua_inc )
+      call get_var5_r4( ncid, 'v_inc', isc,iec, jsc,jec, 1,npz, itile, 1, increment_data%va_inc )
+      call get_var5_r4( ncid, 'T_inc', isc,iec, jsc,jec, 1,npz, itile, 1, increment_data%temp_inc )
+      call get_var5_r4( ncid, 'delp_inc', isc,iec, jsc,jec, 1,npz, itile, 1, increment_data%delp_inc )
+      if ( .not. Atm%flagstruct%hydrostatic ) then
+         call get_var5_r4( ncid, 'delz_inc', isc,iec, jsc,jec, 1,npz, itile, 1, increment_data%delz_inc )
+      end if
+
+      ! Read tracer increments
+      do itracer = 1,ntracers
+         call get_tracer_names(MODEL_ATMOS, itracer, tracer_name)
+         call check_var_exists(ncid, trim(tracer_name)//'_inc', ncerr)
+         if ( ncerr == 0 ) then
+            call get_var5_r4( ncid, trim(tracer_name)//'_inc', isc,iec, jsc,jec, 1,npz, itile, 1, increment_data%tracer_inc(:,:,:,itracer) )
+         end if
+      end do
+         
+      ! Close increment file
+      call close_ncfile(ncid)
```

Comment:
You may get a complaint from operations that this is too disruptive to the filesystem, especially for large numbers of mpi-ranks.  You have every core opening the file, performing reads and then closing the file.  While I know you are using these embedded I/O routines from the dycore, they were originally put in as hacks a long time ago and generally won't play nice now that we've scaled up to thousands or cores on distributed filesystems.

You'd be better off using the fms2_io routines which are already domain aware (suited for native grid) and will read in the correct increment for a given mpi-rank.  The fms2_io routines have recently been modified by UFS contributors to use MPI-IO for reads of NetCDF4 files ([FMS PR#1477](https://github.com/NOAA-GFDL/FMS/pull/1477)).